response_types.go

```diff
@@ -9,27 +9,30 @@ import (
 	"time"
 )
 
-type accessToken struct {
+type AccessToken struct {
```

Comment:
I would create a new interface Token instead of making this struct public. Otherwise we are locking ourselves with OAuth 2 for the foreseeable future.

Suggestion:

```go
interface Token {
	func ExpiresAt() time.Time
	func Value() string
}
```